pkg/runtime/signals/signal_test.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signals
+
+import (
+	"os"
+	"syscall"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("runtime signal", func() {
+
+	Context("SignalHandler Test", func() {
+		It("should close stop channel by signal", func() {
+
+			stop := SetupSignalHandler()
+			p, err := os.FindProcess(syscall.Getpid())
```

Comment:
This is cool, but the handlers aren't setup right it might doing something weird.  It also changes the signal handler for the whole process.

How about

1.  write a small program and exec it
2. mock out the `signal.Notify` and `os.Exit` calls as functions that we re-assign.